pkg/cmd/metrics.go

```diff
@@ -3,141 +3,234 @@ package cmd
 import (
 	"context"
 	"fmt"
-	"io/ioutil"
-	"net/url"
 	"os"
 
-	"github.com/go-kit/log/level"
-	"github.com/observatorium/obsctl/pkg/config"
+	"github.com/observatorium/api/client"
+	"github.com/observatorium/api/client/parameters"
+	"github.com/observatorium/obsctl/pkg/fetcher"
 	"github.com/spf13/cobra"
 )
 
-// TODO(saswatamcode): Add flags for URL query params.
 func NewMetricsGetCmd(ctx context.Context) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "get",
 		Short: "Read series, labels & rules (JSON/YAML) of a tenant.",
 		Long:  "Read series, labels & rules (JSON/YAML) of a tenant.",
 	}
 
+	// Series command.
 	var seriesMatchers []string
+	var seriesStart, seriesEnd string
 	seriesCmd := &cobra.Command{
 		Use:   "series",
 		Short: "Get series of a tenant.",
 		Long:  "Get series of a tenant.",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			u := "/api/v1/series"
-			params := url.Values{}
-
-			for _, s := range seriesMatchers {
-				params.Add("match[]", s)
+			f, currentTenant, err := fetcher.NewCustomFetcher(ctx, logger)
+			if err != nil {
+				return fmt.Errorf("custom fetcher: %w", err)
 			}
 
-			if len(params.Encode()) != 0 {
-				u = u + "?" + params.Encode()
+			params := &client.GetSeriesParams{}
+			if len(seriesMatchers) > 0 {
+				params.Match = (parameters.SeriesMatcher)(seriesMatchers)
+			}
+			if seriesStart != "" {
+				params.Start = (*parameters.StartTS)(&seriesStart)
+			}
+			if seriesEnd != "" {
+				params.End = (*parameters.EndTS)(&seriesEnd)
 			}
 
-			b, err := config.DoMetricsGetReq(ctx, logger, u)
+			resp, err := f.GetSeriesWithResponse(ctx, currentTenant, params)
 			if err != nil {
-				if len(b) != 0 {
-					if perr := prettyPrintJSON(b, cmd.OutOrStdout()); perr != nil {
-						return fmt.Errorf("%v error pretty printing: %v", perr, err)
+				return fmt.Errorf("getting response: %w", err)
+			}
+
+			if resp.StatusCode()/100 != 2 {
+				if len(resp.Body) != 0 {
+					if perr := prettyPrintJSON(resp.Body, cmd.OutOrStdout()); perr != nil {
+						return fmt.Errorf("request failed with statuscode %d pretty printing: %v", perr, resp.StatusCode())
 					}
-					return err
+					return fmt.Errorf("request failed with statuscode %d", resp.StatusCode())
 				}
-				return err
 			}
 
-			return prettyPrintJSON(b, cmd.OutOrStdout())
+			return prettyPrintJSON(resp.Body, cmd.OutOrStdout())
 		},
 	}
 	seriesCmd.Flags().StringArrayVarP(&seriesMatchers, "match", "m", nil, "Repeated series selector argument that selects the series to return.")
+	seriesCmd.Flags().StringVarP(&seriesStart, "start", "s", "", "Start timestamp.")
+	seriesCmd.Flags().StringVarP(&seriesEnd, "end", "e", "", "End timestamp.")
 	err := seriesCmd.MarkFlagRequired("match")
 	if err != nil {
 		panic(err)
 	}
 
+	// Labels command.
+	var labelMatchers []string
+	var labelStart, labelEnd string
```

Comment:
Nit: Similar to previous (and all occurrences below as well), we could declare in one `var`